api/v1alpha1/utils.go

```diff
@@ -37,3 +40,7 @@ func ConditionEquals(existing, expected metav1.Condition) bool {
 
 	return true
 }
+
+func IsValidResourceQuantity(quantity *k8sresource.Quantity) bool {
+	return quantity != nil && quantity.String() != "0" && quantity.String() != ""
```

Comment:
why is "0" not a valid Quantity?

Reply:
Well, the empty value of `resource.Quantity` was coming as `0` so thats why I filtered that out.

Reply:
I have allowed zero value now. We would need to have `defaulting` in api-server level (defaulting webhook etc.). Because if the values are not provided in the CR Spec, then it is "0".